### PR TITLE
Implement context menu UI for choosing package download location

### PR DESCRIPTION
**Pull request by ellensi**
### Purpose

Reference: [MAGN-8192](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8192)

This is the UI sub-task of [MAGN-7766](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7766): As a User, I want to specify the location of Packages at the time of download.

Clicking on a small drop-down button on the right side of an Install button will show a context menu:
- To download the package to the default location
- To choose a custom directory

image

Drop-down context menu for the *default* install button:
- Install latest version
- Install latest version to directory...

Drop-down context menu for the *version-specific* install button:
- Install this version
- Install this version to directory...

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@Benglin 

### FYIs

@lillismith @Racel @riteshchandawar

### CODE REVIEW

**Comment by riteshchandawar** [comment]
@lillismith @Racel @pboyer : Please provide your feedback on Button design and the tool tip text.

Thanks,
Ritesh

**Comment by Racel** [comment]
LGTM - One suggestion, change "directory" to "folder". "Folder" may be a term that users are more familiar with.

File: src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml [hunk 1]
```diff
@@ -492,6 +492,56 @@
...
+
+                        </ControlTemplate>
+                    </Setter.Value>
+                </Setter>
+            </Style>
+
+        </Style.Resources>
+
+    </Style>
+
```
Comment (Benglin) [inline src/DynamoCoreWpf/UI/Themes/Modern/DynamoModern.xaml]: If this `Style` isn't used anywhere else than in `PackageManagerSearchView.xaml`, then there's no reason to add to the cost of start-up. Can we move this style into `PackageManagerSearchView.xaml` so it's only parsed once when that dialog is shown?

File: src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml.cs [hunk 2]
```diff
@@ -31,11 +32,28 @@ public void ItemStackPanel_MouseDown(object sender, RoutedEventArgs e)
             viewModel.Model.IsExpanded = !viewModel.Model.IsExpanded;
```
Comment (Benglin) [inline src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml.cs]: For consistency, please rename the following event handlers to:

`OnShowContextMenuFromLeftClicked`
`OnSortButtonClicked`
`OnInstallLatestButtonDropDownClicked`
`OnInstallVersionButtonDropDownClicked`

File: src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml [hunk 3]
```diff
@@ -285,12 +299,26 @@
...
+                                                                        Click="InstallVersionButtonDropDown_OnClick">
+                                                                    <Button.ContextMenu>
+                                                                        <ContextMenu>
+                                                                            <MenuItem Header="{x:Static p:Resources.PackageSearchViewInstallThisVersion}" Command="{Binding Path=Item2}" />
+                                                                            <MenuItem Header="{x:Static p:Resources.PackageSearchViewInstallThisVersionTo}" />
+                                                                        </ContextMenu>
+                                                                    </Button.ContextMenu>
+                                                                </Button>
+                                                            </StackPanel>
+                                                        </Border>
```
Comment (Benglin) [inline src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml]: While you're at it, can you do a document-wide auto formatting by `CTRL+K+D`?

**Comment by Benglin** [comment]
Thanks @ellensi, few more tweaks and we'll be good to go!

**Comment by Benglin** [comment]
Thanks for the tweaks, @ellensi!

**Pull request merged**